atomic_reactor/plugin.py

```diff
@@ -211,6 +225,7 @@ def run(self, keep_going=False):
             except Exception as ex:
                 msg = "plugin '%s' raised an exception: %r" % (plugin_instance.key, ex)
                 logger.debug(traceback.format_exc())
+                self.save_plugin_error(plugin_instance.key, repr(ex))
```

Comment:
The thing I was hoping we could track as far as exceptions go is: "which exception caused the build to fail, and which plugin did it come from?"

What's being stored here is a slightly different thing: "which plugins raised which exceptions?" It doesn't tell us which one actually caused the build to fail -- not all exceptions cause failures.

So I think we should be saving this error in the `on_plugin_failed()` section further down instead.